lib/search-field/index.jsx

```diff
@@ -27,13 +27,19 @@ export class SearchField extends Component {
   };
 
   componentDidUpdate() {
-    const { searchFocus, onSearchFocused } = this.props;
+    const { searchFocus, onSearchFocused, filter } = this.props;
 
     if (searchFocus && this.inputField) {
       this.inputField.select();
       this.inputField.focus();
       onSearchFocused();
     }
+
+    // check to see if the filter has been updated (by a tag being clicked from suggestions)
+    // this is a hack to work around query not being in app state (yet)
+    if (filter !== this.state.query) {
+      this.inputField.value = filter;
```

Comment:
we can probably also accomplish this here by manually updating local state…

```js
if ( filter !== this.state.query ) {
	this.debouncedSearch.cancel();
	this.setState( { query: filter } );
}
```

Reply:
Hahaha I tried exactly this first and it destroyed the universe.

Reply:
No wait, maybe I tried .flush (which is what clear search does). Lemme see...

Reply:
`flush()` runs the currently-queued function while `cancel()` prevents it from happening

if we did `flush()` it would try and update the search filter and immediately clear out the results then set state and reset the search field, I think

Reply:
Fascinatingly, if I do `.cancel` and `setState` instead I am unable to type anything at all in the search field.

Reply:
oh wow. that makes sense now because I see that `.cancel` stops the entire debounced function when we only want to stop a specific invocation.

this probably all sounds more complicated than it needs to be

```js
if ( filter !== this.state.query ) {
	this.debouncedSearch( filter );
  this.debouncedSearch.flush();
}
```

^^^ how about that? update the search through the normal means but also immediately flush it to commit those changes

Reply:
Doesn't work either :/